mage_integrations/mage_integrations/destinations/redshift/__init__.py

```diff
@@ -132,31 +142,40 @@ def build_insert_commands(
 
         if unique_constraints and UNIQUE_CONFLICT_METHOD_UPDATE == unique_conflict_method:
             full_table_name_temp = self.full_table_name(schema_name, table_name, prefix='temp_')
-            full_table_name_old = self.full_table_name(schema_name, table_name, prefix='old_')
             drop_temp_table_command = f'DROP TABLE IF EXISTS {full_table_name_temp}'
-            drop_old_table_command = f'DROP TABLE IF EXISTS {full_table_name_old}'
             unique_constraints_clean = [
                 f'{self.clean_column_name(col)}'
                 for col in unique_constraints
             ]
+
+            drop_duplicate_records_from_temp = (f'DELETE FROM {full_table_name_temp} '
+            f'WHERE ({", ".join(unique_constraints_clean)},_mage_created_at) '
+            f'IN ( SELECT {", ".join(unique_constraints_clean)}, _mage_created_at '
+            f'FROM ( SELECT {", ".join(unique_constraints_clean)}, _mage_created_at, ROW_NUMBER() '
+            f'OVER (PARTITION BY {", ".join(unique_constraints_clean)} '
+            f'ORDER BY _mage_created_at DESC) as row_num '
+            f'FROM {full_table_name_temp} ) AS subquery_alias WHERE row_num > 1);')
+
+            delete_records_from_full_table = (f'DELETE FROM {full_table_name} '
+                                              f'WHERE ({", ".join(unique_constraints_clean)}) '
+                                              f'IN (SELECT {", ".join(unique_constraints_clean)} '
+                                              f'FROM {full_table_name_temp})')
+
+            insert_records_from_temp_table = f'INSERT INTO {full_table_name} SELECT * FROM {full_table_name_temp}'
+            truncate_records_from_temp_table = f'TRUNCATE TABLE {full_table_name_temp}'
+
+            commands = [
+                '\n'.join([
+                    f'INSERT INTO {full_table_name_temp} ({insert_columns})',
+                    f'VALUES {insert_values}',
+                ]),
+            ]
+            # This is temp as need to know the best way to create table programmatically i will change it properly
```

Comment:
can you elaborate on this comment? not sure what you mean here

Reply:
yeah, my bad will remove this comment